### ISSUE TITLE
Permalink to application form

### ISSUE THREAD

**Issue by elliottwilliams** [matchboxcowork/matchboxstudio.org#15]
https://matchboxstudio.org/apply should load the application form.

Technically, I think this could be implemented a few different ways:
- **Single-page application**: We can continue building the site as a single-page webapp. Then, nginx can be configured to redirect every non-php request to `index.html`, and a dispatcher on the page can load the appropriate resource.
- **Multi-page site**: We could move the application to its own page. The trick here would be to still embed it in `index.html` without having to repeat ourselves. This would probably entail creating a partial file of the form itself and building a container page.
- **Just redirect**: We could redirect to `index.html#apply` and have some JS on the page expand the application and scroll down appropriately.

Currently, I think the second approach is best, especially when considering how I'd like build the about page in #13.

**Comment by jhewitt12** [comment]
For the multi-page site option, can you explain that process a little more. I'm new to haml. So I created a new haml file in the partials folder. How do I include it in index.html and make it its own page?

**Comment by elliottwilliams** [comment]
Sure (btw, I highly reccommend http://haml.info/docs.html if you're still getting used to the language).

The initial problem I was trying to address here was an answer to the question "Can we get a permalink to the home page". The underlying issue is that the application form is hidden by default, inside `index.haml`. What's the easiest way to keep that hidden-by-default form while making it linkable?

My idea for the multi-page approach would be to actually move the application form to it's own page (say, `apply.haml`). Then we could link directly to that page to get to the apply form. On `index.haml`, we'd still need to embed the form.

Partial files are reusable components that can be used on multiple pages. So, in order to put the same application form on `index.haml`, and `apply.haml`, we could put it inside a partial, then include the partial on both of those pages. It'd look something like this:

partials/application_form.haml:
```haml
%form.application-form
   %input{title: "Name"}
   ...
   %input{type: "submit"}
```

index.haml:
```haml
%html
  %body
    ...
    %section.application-form-container.hidden-by-default
      = include 'partials/application_form.haml'
    ...
```

apply.haml
```haml
 %html
  %body
    = include 'partials/application_form.haml'
    ...
```

If you want to understand the `include` function better, check out its definition in https://github.com/matchboxcowork/matchboxstudio.org/blob/master/haml_extensions.rb#L29. It's quite simple: it loads the template file passed to it, asks haml to render that file, and returns the rendered result to be included in the resulting page.

**Comment by elliottwilliams** [comment]
Also, keep in mind that this ticket will be closed when my video hero pull request (#29) is merged. See [my comment there](https://github.com/matchboxcowork/matchboxstudio.org/pull/29#issue-191326661): 

> I also reimplemented the way the membership application is opened, so once this is merged, we can direct-link to the application from http://matchboxstudio.org/#app (closing #15).

So, this issue can be closed once that change is rolled out. Incidentally, my change takes the third approach ("Just redirect") to solving this problem.
